scylla/src/statement/prepared_statement.rs

```diff
@@ -19,6 +19,7 @@ pub struct PreparedStatement {
     metadata: PreparedMetadata,
     statement: String,
     page_size: Option<i32>,
+    partitioner_name: Option<String>,
```

Comment:
Once we have an enum like
```
pub enum PartitionerName {
    Murmur3,
    CDC,
}
```
what should be the type of this field? There are two possibilities:

- `PartitionerName` with implemented `Default` as `PartitionerName::Murmur3`. The drawback is that it hardcodes "the defaultability" of Murmur3 in the "wrong" place - I don't know how to properly express it but it kinda feels wrong.
- `Option<PartitionerName>`. The drawback is that we have to duplicate the logic handling empty partitioner. First we need to map from string to the field itself when statement is prepared with something like
```
self.partitioner_name = Some(match partitioner_name {
            Some(partitioner_name) if partitioner_name.ends_with("CDCPartitioner") => {
                PartitionerName::CDC
            }
            _ => PartitionerName::Murmur3,
        })
```
and when the statement is executed and we need to actually pick the correct partitioner we need to do something like
```
match partitioner_name {
            None | Some(PartitionerName::Murmur3) => Ok(Murmur3Partitioner::hash(partition_key)),
            Some(PartitionerName::CDC) => Ok(CDCPartitioner::hash(partition_key)),
        }
```
The logic is more or less repeated here and this could be avoided if we chose the first aproach.

What do you think?

Reply:
I think the first option is fine. The Murmur3 partitioner _is_ the default partitioner for Scylla. If we keep this type as internal for the crate - not `pub`, at most `pub(crate)` - I don't think it hurts making Murmur3 the `Default`.